src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -58,9 +43,30 @@ public static string GetFullPath(string path, string basePath)
                 throw new ArgumentException(SR.Argument_InvalidPathChars);
 
             if (IsPathFullyQualified(path))
-                return GetFullPath(path);
+                return GetFullPathInternal(path);
+
+            return GetFullPathInternal(CombineInternal(basePath, path));
+        }
 
-            return GetFullPath(CombineInternal(basePath, path));
+        // Gets the full path without argument validation
+        private static string GetFullPathInternal(string path)
+        {
```

Comment:
```suggestion
        {
            Debug.Assert(!string.IsNullOrEmpty(path));
            Debug.Assert(!path.Contains('\0'));
            
```
to make it clear that these conditions are expected to be validated by the caller.